### PR TITLE
dcos-net-errors checker that counts errors on dcos-net logs

### PR DESCRIPTION
**Pull request by Fabs**
- Adds a checker that counts errors on dcos-net logs according to the regex: "[error]|[critical]|[alert]|[emergency]"

### CODE REVIEW

File: checks/networking/dcos_net_errors.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	"regexp"
+	"github.com/adyatlov/bun"
+)
+// http://erlang.org/doc/apps/kernel/logger_chapter.html#log_level
+var errorsRegex = "\\[error\\]|\\[critical\\]|\\[alert\\]|\\[emergency\\]"
+
+func init() {
+	builder := bun.CheckBuilder{
+		Name: "net-errors-checker",
+		Description: "Identify errors on dcos-net logs",
```
Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: nit: in dcos-net logs?

File: checks/networking/dcos_net_errors.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		CollectFromAgents:       collect,
+		CollectFromPublicAgents: collect,
+		Aggregate:               aggregate,
+	}
+	check := builder.Build()
+	bun.RegisterCheck(check)
+}
+
+func collect(host bun.Host) (ok bool, details interface{}, err error) {
+	errorMatcher, _ := regexp.Compile(errorsRegex)
```
Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: How about using `MustCompile` here?

File: checks/networking/dcos_net_errors.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  keys := 0
```
Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: nit: bad indentation

File: checks/networking/dcos_net_errors.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+	file, err1 := host.OpenFile("net")
+
+	if err1 != nil {
+		ok = false
+		errMsg := fmt.Sprintf("Cant open file %s", err1)
```
Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: nit: `cannot open net file`

File: checks/networking/dcos_net_errors.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+	}
+	check := builder.Build()
+	bun.RegisterCheck(check)
+}
+
+func collect(host bun.Host) (ok bool, details interface{}, err error) {
+	errorMatcher, _ := regexp.Compile(errorsRegex)
+  keys := 0
+
+	file, err1 := host.OpenFile("net")
```
Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: nit: why `err1` with `1` in the end?

Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: Is it intentionally non-configurable?

File: checks/networking/dcos_net_errors.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		ok = false
+		errMsg := fmt.Sprintf("Cant open file %s", err1)
+		fmt.Println(errMsg)
+		return
+	}
+
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+  for scanner.Scan() {
```
Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: bad indentation

File: checks/networking/dcos_net_errors.go [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+		line := scanner.Text()
+		if errorMatcher.MatchString(line) {
+			keys++
+		}
+  }
```
Comment (ichernetsky) [inline checks/networking/dcos_net_errors.go]: ditto

Comment (Fabs) [reply checks/networking/dcos_net_errors.go]: I have installed some tooling to help me with this, there was mixed tabs and spaces. It looked good on emacs but it was wrong. Sorry for that.

File: checks/networking/dcos_net_errors.go (hunk 5 above)
Comment (Fabs) [reply checks/networking/dcos_net_errors.go]: There is a file who maps "net" to the actual diagnostics bundle file `filetypes/files_type_yaml.go`. In what sense you want this to be configured? A list of files with a single one maybe?

**Pull request closed without merging**